src/server/src/modules/user.ts

```diff
@@ -48,8 +51,32 @@ export const createUser = (
         firstName,
         lastName,
         role,
+        emailVerified: false
     });
-    return userToInsert.save();
+
+    await userToInsert.save();
+
+    let verificationKeyString: string;
+
+    do {
+        verificationKeyString = makeRandomString(/*length=*/20);
+    } while (await VerificationKey.findOne({ verificationKey: verificationKeyString }));
+
+    const verificationKey = VerificationKey.create({
+        user: userToInsert,
+        verificationKey: verificationKeyString
+    });
+
+    await verificationKey.save();
+    
+    // Send email with the verification link.
+    sendEmail(
+        email,
+        `Verify your email address`,
+        `Welcome to R'match! Please verify your email address so we know that it's you. http://localhost:3000/verify/${verificationKeyString}`,
```

Comment:
The body of the email could be something like this:

Welcome to R'Match!

Your account has been created. Please follow this link: *your link* to verify your email and complete your registration.

Reply:
Done.